out/notebooks/spl2/installer.ts

```diff
@@ -0,0 +1,506 @@
+    );
+    return binJavaPath;
+}
+
+async function promptToDownloadLsp(alsoInstallJava: boolean): Promise<boolean> {
+    const promptMessage = (
+        'For SPL2 support with this extension a download of the SPL2 Language ' +
+        'Server subject to the Splunk General Terms ' +
+        (
+            alsoInstallJava ?
+                `and of Java ${minimumMajorJavaVersion} are required.` :
+                'is required.'
+        )
+    );
+    const agreeAndContinueChoice = 'Agree and Continue';
+    const viewTermsChoice = 'View Splunk General Terms';
+    const turnOffSPL2Choice = 'Turn off SPL2 support';
+
+    const popup = window.showInformationMessage(
+        promptMessage,
+        { modal: true },
+        agreeAndContinueChoice,
+        viewTermsChoice,
+        turnOffSPL2Choice,
+    );
+    
+    const userSelection = (await popup) || null;
+    switch(userSelection) {
+        case agreeAndContinueChoice:
+            // Record preference so user is not asked again
+            workspace.getConfiguration().update(configKeyAcceptedTerms, TermsAcceptanceStatus.Accepted, true);
+            return true;
+        case viewTermsChoice:
+            console.log('Viewing Splunk General Terms in browser...');
+            env.openExternal(Uri.parse('https://www.splunk.com/en_us/legal/splunk-general-terms.html'));
+            return await promptToDownloadLsp(alsoInstallJava);
+        case turnOffSPL2Choice:
+            console.log('User opted out of SPL2 Langauge Server download, SPL2 support disabled');
+            // Record preference so user is not asked again
+            workspace.getConfiguration().update(configKeyAcceptedTerms, TermsAcceptanceStatus.DeclinedForever, true);
+            return false;
+        default:
+            // Cancel
+            return false;
+    }
+}
+
+/**
+ * Checks if the installed SPL2 Language Server version is the latest and prompt for
+ * upgrade if not, or automatically upgrade if user has that setting enabled.
+ */
+export async function getLatestSpl2Release(context: ExtensionContext, progressBar: StatusBarItem): Promise<void> {
+    return new Promise(async (resolve, reject) => {
+        const lspArtifactPath = path.join(context.globalStorageUri.fsPath, 'lsp');
+        // TODO: Remove this hardcoded version/update time and check for updates
+        let lspVersion: string = '2.0.362'; // context.globalState.get(stateKeyLatestLspVersion) || "";
+        const lastUpdateMs: number = Date.now(); // context.globalState.get(stateKeyLastLspCheck) || 0;
+        // Check for new version of SPL2 Language Server if longer than 24 hours
+        if (Date.now() - lastUpdateMs > 24 * 60 * 60 * 1000) {
+            const metaPath = path.join(lspArtifactPath, 'maven-metadata.xml');
+            try {
+                await downloadWithProgress(
+                    'https://splunk.jfrog.io/splunk/maven-splunk-release/spl2/com/splunk/spl/spl-lang-server-sockets/maven-metadata.xml',
```

Comment:
For some reason I'm having trouble getting axios to download this file and receive a `close` event from the data stream. Once I get this working we can check for the latest version of the language server and automatically download new versions but for now we can hardcode the version to be used in VSCode.